File: src/main/java/com/tobe/healthy/common/CustomPagingResponse.java
package com.tobe.healthy.common;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CustomPagingResponse<T> {

	@JsonInclude(Include.NON_NULL)
	private String studentName;

	private List<T> content;

	private int pageNumber;

	private int pageSize;

	private int totalPages;

	private long totalElements;

	@JsonProperty("isLast")
	private boolean isLast;
}

File: src/main/kotlin/com/tobe/healthy/common/CustomPagingResponse.kt
package com.tobe.healthy.common

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL

data class CustomPagingResponse<T>(
    @JsonInclude(NON_NULL)
    val studentName: String? = null,
    val content: List<T>,
    val pageNumber: Int,
    val pageSize: Int,
    val totalPages: Int,
    val totalElements: Long,
    val isLast: Boolean
)
